### ISSUE TITLE
ImagePullBackOff for images on local environment

### ISSUE THREAD

**Issue by petr-buchin** [GoogleContainerTools/skaffold#3310]
### Expected behavior
We have a really big platform, created using SOA. 
It has many services, and all that services could not be launched on a single developer machine.
Therefore we use Skaffold profiles to launch separate projects.

We also have separate Skaffold profiles for all the artifacts, in order to be able to run local builds in parallel processes, something like that:

```
skaffold build -p=artifact-a --cache-artifacts=true
skaffold build -p=artifact-b --cache-artifacts=true
skaffold build -p=artifact-c --cache-artifacts=true
```

After all artifacts are built I include all needed profile names in skaffold command, like follows:

```
skaffold dev -p=artifact-a,artifact-b,artifact-c,service-a,service-b,service-c --cache-artifacts=true
```

We use Helm as our deployer, and command above works and creates needed `k8s` resources.

However, all our pods aren't running and have state `ImagePullBackOff`.
`kubectl describe pod` shows that pods are trying to pull images with `latest` tag instead of tag, generated by Skaffold. 
At the same time `docker image ls` shows that we have all image tags, created by Skaffold.

Thus, **expected behavior** is for pods to use skaffold-generated image tags.

### Actual behavior
Pods use `latest` image tags

### Information

- Skaffold version: 1.0.1
- MacOs Catalina
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta16
kind: Config
metadata:
  name: platform
build:
  local:
    push: false
  artifacts: []
deploy:
  helm:
    releases:
      - name: example-release
        chartPath: charts/example
        valuesFiles:
        - values-file-a.yaml
        - values-file-b.yaml
        values: {}
profiles:
- name: artifact-a
  patches:
  - op: add
    path: /build/artifacts
    value:
    - image: example/artifact-a
      context: artifact-a
      docker:
        dockerfile: Dockerfile
  - op: add
     path: /deploy/helm/releases/0/values
     value:
       subchart-a.image: example/artifact-a
- name: artifact-b
  patches:
  - op: add
    path: /build/artifacts
    value:
    - image: example/artifact-b
      context: artifact-b
      docker:
        dockerfile: Dockerfile
  - op: add
    path: /deploy/helm/releases/0/values
    value:
      subchart-b.image: example/artifact-b
- name: service-a
  patches:
  - op: add
    path: /deploy/helm/releases/0/valuesFiles/-
    value: charts/example/values/_services/service-a.yaml # enables some subchart
- name: service-b
  patches:
  - op: add
    path: /deploy/helm/releases/0/valuesFiles/-
    value: charts/example/values/_services/service-b.yaml # enables some subchart

```

**Comment by dgageot** [comment]
@petr-buchin thanks a lot for sharing your interesting use case. Before I try to reproduce, here's some feedback:
 + `--cache-artifacts=true` is the default so you should be able to remove this flag
 + You can replace
```- op: add
    path: /build/artifacts
    value:
    - image: example/artifact-b
      context: artifact-b
      docker:
        dockerfile: Dockerfile
```
with
```- op: add
    path: /build/artifacts
    value:
    - image: example/artifact-b
      context: artifact-b
```

**Comment by dgageot** [comment]
could you share the output of `skaffold run -vdebug`?

**Comment by petr-buchin** [comment]
Hi @dgageot , thanks a lot for a quick reply and for advices!

Here is the output of `skaffold run -vdebug`:
```
INFO[0000] Skaffold &{Version:580a176 ConfigVersion:skaffold/v2alpha1 GitVersion: GitCommit:580a1764476a11a7130b794825cbba63d3ad828d GitTreeState:clean BuildDate:2019-11-26T23:18:22Z GoVersion:go1.12 Compiler:gc Platform:darwin/amd64} 
DEBU[0000] config version (skaffold/v1beta16) out of date: upgrading to latest (skaffold/v2alpha1) 
DEBU[0000] Defaulting build type to local build         
DEBU[0000] Defaulting deploy type to kubectl            
DEBU[0000] validating yamltags of struct SkaffoldConfig 
DEBU[0000] validating yamltags of struct Metadata       
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct GitTagger      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct LocalBuild     
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct KubectlDeploy  
DEBU[0000] validating yamltags of struct KubectlFlags   
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct Artifact       
DEBU[0000] validating yamltags of struct ArtifactType   
DEBU[0000] validating yamltags of struct CustomArtifact 
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct LocalBuild     
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct HelmDeploy     
DEBU[0000] validating yamltags of struct HelmRelease    
DEBU[0000] validating yamltags of struct HelmOverrides  
DEBU[0000] validating yamltags of struct HelmImageStrategy 
DEBU[0000] validating yamltags of struct HelmImageConfig 
DEBU[0000] validating yamltags of struct HelmDeployFlags 
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
DEBU[0000] validating yamltags of struct Profile        
DEBU[0000] validating yamltags of struct Pipeline       
DEBU[0000] validating yamltags of struct BuildConfig    
DEBU[0000] validating yamltags of struct TagPolicy      
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct JSONPatch      
DEBU[0000] validating yamltags of struct YamlpatchNode  
DEBU[0000] validating yamltags of struct Node           
INFO[0000] Using kubectl context: docker-desktop        
DEBU[0000] Using builder: local                         
DEBU[0000] setting Docker user agent to skaffold-580a176 
DEBU[0000] push value not present, defaulting to false because localCluster is true 
Generating tags...
INFO[0000] Tags generated in 45.401µs                   
Checking cache...
INFO[0000] Cache check complete in 141.193µs            
Tags used in deployment:
Starting deploy...
DEBU[0000] Running command: [kubectl version --client -ojson] 
DEBU[0000] Command output: [{
  "clientVersion": {
    "major": "1",
    "minor": "16",
    "gitVersion": "v1.16.2",
    "gitCommit": "c97fe5036ef3df2967d086711e6c0c405941e14b",
    "gitTreeState": "clean",
    "buildDate": "2019-10-15T23:41:55Z",
    "goVersion": "go1.12.10",
    "compiler": "gc",
    "platform": "darwin/amd64"
  }
}
] 
WARN[0000] k8s/*.yaml did not match any file            
INFO[0000] Deploy complete in 218.999397ms              
You can also run [skaffold run --tail] to get the logs
```

**Comment by petr-buchin** [comment]
@dgageot  I actually found the reason for this behavior.
It's the  `yaml:"artifacts,omitempty"` tag in `BuildConfig` struct, which makes my root build config field `artifacts: []` dissapear during Skaffold config unmarshaling.

As a result of this field ommitting, I cannot use path `/build/artifacts/-` for patches, so I used `/build/artifacts`, which resulted in a total replacement of the whole `artifacts` sequence after applying every patch.

Could we please delete `omitempty` for `build.artifacts` and `helm.releases.*.values` fields and allow users to specify empty arrays/maps at root level and patch them in profiles?

**Comment by nkubala** [comment]
ugh, yeah I really hate the fact that this is happening. we use `omitempty` as a yamltag so we don't generate nonsense for people in `skaffold init`, but it does indeed create weirdness when trying to apply profiles to fields that aren't specified. I dug deep into this in another issue and explained it here: https://github.com/GoogleContainerTools/skaffold/issues/2741#issuecomment-571799226

one workaround for this is supplying a "dummy" value in your default config that is always overridden by the profile values, but this is obviously not great.

we've been talking about ways we can improve our profile patching, and one idea would be to use kustomize as the default for templating out configuration in the skaffold.yaml at runtime. I don't have an answer on what we'll be doing yet, but stay tuned for more developments there.

in the meantime I'll close this issue since it's being tracked elsewhere.
